### PR TITLE
Use the back stack to determine nav top level ownership

### PR DESCRIPTION
**Pull request by tunjid**
Addresses #867.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ androidxHiltNavigationCompose = "1.0.0"
 androidxLifecycle = "2.6.1"
 androidxMacroBenchmark = "1.2.0-alpha16"
 androidxMetrics = "1.0.0-alpha03"
-androidxNavigation = "2.5.3"
+androidxNavigation = "2.6.0"
```
Comment (alexvanyo) [inline gradle/libs.versions.toml:19]: Any reason not to bump to latest stable `2.7.0`?

File: app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt [hunk 2]
```diff
@@ -97,6 +105,37 @@ class NiaAppState(
             else -> null
         }
 
+    val backStackEntries: List<NavBackStackEntry>
+        @Composable get() {
+            // TODO: Read backStack directly from the navController when b/295553995 is resolved.
```
Comment (alexvanyo) [inline app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt:110]: Nit: Can this be the public issue link? https://issuetracker.google.com/issues/295553995

**Review by alexvanyo — COMMENTED** [review]
Two nits, and can we add a regression test verifying that #867 is fixed?

**Comment by tunjid** [comment]
Will need to rework this, the better solution is to track the last tab tapped to and use that to drive the indicator.

**Pull request closed without merging**